src/exchange_price_source.ts

```diff
@@ -0,0 +1,130 @@
+import { ExchangeAdapter, Ticker } from './exchange_adapters/base'
+import BigNumber from 'bignumber.js'
+import { PriceSource, WeightedPrice } from './price_source'
+import { Exchange, OracleCurrencyPair } from './utils'
+import { MetricCollector } from './metric_collector'
+import { individualTickerChecks } from './aggregator_functions'
+
+export interface OrientedExchangePair {
+  exchange: Exchange
+  symbol: OracleCurrencyPair
+  to_invert: boolean
+}
+
+export interface ExchangePriceSourceConfig {
+  pairs: OrientedExchangePair[]
+}
+
+export type OrientedAdapter = [ExchangeAdapter, boolean]
```

Comment:
Imo this should be an object instead of an array just for extra clarity

Reply:
I meant it as a tuple. Changed it to an object.